controllers/watermarkpodautoscaler_controller.go

```diff
@@ -355,11 +358,11 @@ func shouldScale(logger logr.Logger, wpa *datadoghqv1alpha1.WatermarkPodAutoscal
 		transitionCountdown.With(prometheus.Labels{wpaNamePromLabel: wpa.Name, transitionPromLabel: "upscale", resourceNamespacePromLabel: wpa.Namespace, resourceNamePromLabel: wpa.Spec.ScaleTargetRef.Name, resourceKindPromLabel: wpa.Spec.ScaleTargetRef.Kind}).Set(0)
 	}
 
-	return canScale(logger, backoffUp, backoffDown, currentReplicas, desiredReplicas)
+	return canScale(logger, backoffUp, backoffDown, currentReplicas, desiredReplicas, wpa)
```

Comment:
since you added the logs attributes in the `reconcile()` function.
you don't need anymore to provide the `wpa` to this function